grapher/controls/Controls.tsx

```diff
@@ -237,52 +235,103 @@ export class FilterSmallCountriesToggle extends React.Component<{
 }
 
 export interface FacetStrategyDropdownManager {
-    availableFacetStrategies?: FacetStrategy[]
-    facet?: FacetStrategy
-    showFacets?: boolean
+    availableFacetStrategies: FacetStrategy[]
+    facetStrategy?: FacetStrategy
+    hideFacetControl?: boolean
 }
 
-const FACET_DROPDOWN_CLASS = "FacetStrategyDropdown"
-
+// A drop-down button that, when clicked, shows a hovering visual display
+// indicating the faceting options.
 export class FacetStrategyDropdown extends React.Component<{
     manager: FacetStrategyDropdownManager
 }> {
-    @computed get options(): { label: string; value: string }[] {
-        const strategies = this.props.manager.availableFacetStrategies || [
-            FacetStrategy.none,
-            FacetStrategy.entity,
-            FacetStrategy.column,
-        ]
-        return strategies.map((value) => {
-            const label = facetStrategyLabels[value]
-            return { label, value }
-        })
+    render(): JSX.Element {
+        return (
+            <Tippy
+                content={this.content}
+                interactive={true}
+                trigger={"click"}
+                placement={"bottom-start"}
+                arrow={false}
+            >
+                <div
+                    className="FacetStrategyDropdown"
+                    style={{ width: this.dropDownWidth }}
+                >
+                    {facetStrategyLabels[this.facetStrategy]}
+                    <div>
+                        <FontAwesomeIcon icon={faChevronDown} />
+                    </div>
+                </div>
+            </Tippy>
+        )
+    }
+
+    @computed get strategies(): FacetStrategy[] {
+        return (
+            this.props.manager.availableFacetStrategies || [
+                FacetStrategy.none,
+                FacetStrategy.entity,
+                FacetStrategy.column,
+            ]
+        )
     }
 
-    @computed get facet(): FacetStrategy {
-        return this.props.manager.facet || FacetStrategy.none
+    @computed get dropDownWidth(): string {
+        const maxWidth = Math.max(
+            ...this.strategies.map(
+                (s) =>
+                    Bounds.forText(facetStrategyLabels[s], { fontSize: 12 })
+                        .width
+            )
+        )
+        return `${maxWidth + 45}px` // leave room for the chevron
     }
 
-    @action.bound onChange(
-        selected: ValueType<{ label: string; value: string }>
-    ): void {
-        this.props.manager.facet = asArray(selected)[0].value as FacetStrategy
+    // A hovering visual display giving options to be selected from
+    @computed get content(): JSX.Element {
+        const parts = this.strategies.map((value: FacetStrategy) => {
+            // There are only ever two strategies, none + (country OR entity).
+            const label = facetStrategyLabels[value]
+            const selected = value == this.facetStrategy ? " selected" : ""
```

Comment:
There is something not going quite right with the reactivity, the selected option doesn't correctly propagate: 

https://user-images.githubusercontent.com/1308115/128192216-23e73e63-ec6f-422d-a7e3-dee5179e8348.mov

Reply:
Turns out I forgot to mark the dropdown class with `@observer`, doing that fixes it.